### PR TITLE
Only increment Enable status for RGB Matrix if it supports it

### PR DESCRIPTION
**Pull request by drashna**
## Description

Right now, the RGB Matrix toggle code increments the variable. That's fine, if `RGB_MATRIX_EXTRA_TOG` is defined.  But it's not normally.  So, on non-massdrop boards, it's creating an issue where you have to hit the toggle multiple times. 

This should fix the issue.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
   - [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by XScorpion2** [comment]
Also fixed by https://github.com/qmk/qmk_firmware/pull/5619

File: quantum/rgb_matrix.c [hunk 1]
```diff
@@ -511,7 +511,12 @@ void rgb_matrix_set_suspend_state(bool state) {
 }
 
 void rgb_matrix_toggle(void) {
+#ifdef RGB_MATRIX_EXTRA_TOG
   rgb_matrix_config.enable++;
+#else
+  rgb_matrix_config.enable ^= 1;
+  if (rgb_matrix_config.enable > 1) { rgb_matrix_config.enable = 0; } // make sure that if we are treating this as a bool, that it is only 1 or 0.
```
Comment (yanfali) [inline quantum/rgb_matrix.c]: Format it correctly. You're messing with readability for no real advantage.

Comment (drashna) [reply quantum/rgb_matrix.c]:
```suggestion
  if (rgb_matrix_config.enable > 1) {  // make sure that if we are treating this  as a bool, that it is only 1 or 0.
    rgb_matrix_config.enable = 0; 
  }
```

**Comment by mechmerlin** [comment]
This worked on my doro67, would like to see confirmation that this doesn't break functionality on massdrop boards.

**Comment by drashna** [comment]
@mechmerlin The Massdrop Boards are the only ones with the "extra toggle" defined.   So, yeah, it shouldn't break anything. It just wasn't included in the rgb matrix code, accidently. 

https://github.com/qmk/qmk_firmware/blob/master/keyboards/massdrop/alt/config.h#L126
https://github.com/qmk/qmk_firmware/blob/master/keyboards/massdrop/ctrl/config.h#L126

**Comment by drashna** [comment]
superseded.

**Pull request closed without merging**